tfhe/src/integer/server_key/mod.rs

```diff
@@ -231,6 +231,26 @@ impl ServerKey {
 
         num_bits_to_represent_output_value.div_ceil(num_bits_in_message as usize)
     }
+
+    /// Returns how many ciphertext can be summed at once
+    ///
+    /// The number of ciphertext that can be added together depends on the degree
+    /// (in order not to go beyond the carry space and keep results correct) but also
+    /// on the noise level (in order to have the correct error probability and so correctness and
+    /// security)
+    ///
+    /// - `degree` is expected degree of all elements to be summed
+    pub(crate) fn max_sum_size(&self, degree: Degree) -> usize {
+        let max_degree =
+            MaxDegree::from_msg_carry_modulus(self.message_modulus(), self.carry_modulus());
+        let max_sum_to_full_carry = max_degree.get() / degree.get();
+
+        // We want to compare with the true max noise level, that is the one
+        // were we don't need enough room to accept a carry
+        let true_max_noise_level = max_degree.get() / (self.message_modulus().0 - 1);
```

Comment:
self.key.max_noise_level I believe has the value you are looking for

Reply:
No because its made to be able to receive a carry, when this is for when we want to be completely full

Reply:
not sure I follow, 15 / 3 = 5 and we have 5 in the key ?

Reply:
No, the max degree in the key is computed via

```
  pub(crate) fn integer_radix_server_key(
        message_modulus: MessageModulus,
        carry_modulus: CarryModulus,
    ) -> Self {
        let full_max_degree = message_modulus.0 * carry_modulus.0 - 1;

        let carry_max_degree = carry_modulus.0 - 1;

        // We want to be have a margin to add a carry from another block
        Self::new(full_max_degree - carry_max_degree)
    }
```

Reply:
what I mean is that the noise is dependeant on the parameters, the max degree you already re compute it

Reply:
I yeah bit TRUE, I misread sorry

Reply:
no worries :)